app/src/main/java/pm/gnosis/heimdall/ui/addressbook/detail/AddressBookEntryDetailsActivity.kt

```diff
@@ -44,8 +41,8 @@ class AddressBookEntryDetailsActivity : BaseActivity() {
         super.onCreate(savedInstanceState)
         inject()
 
-        intent.extras.getString(EXTRA_ADDRESS_ENTRY).hexAsEthereumAddressOrNull().let {
-            if (it == null) {
+        (intent.extras.getSerializable(EXTRA_ADDRESS_ENTRY) as? BigInteger?).let {
```

Comment:
I was more thinking of using the BigInteger only in the method signature and than convert it to a hex string to put it in a bundle (not sure about serializable)